### PR TITLE
Delete project member when delete project

### PR DESCRIPTION
**Pull request by stonezdj**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Review by chlins — APPROVED** [review]
lgtm

File: src/pkg/member/manager.go [hunk 1]
```diff
@@ -101,6 +103,9 @@ func (m *manager) Delete(ctx context.Context, projectID int64, memberID int) err
 func (m *manager) DeleteMemberByUserID(ctx context.Context, uid int) error {
 	return m.dao.DeleteProjectMemberByUserID(ctx, uid)
 }
+func (m *manager) DeleteMemberByProjectID(ctx context.Context, projectID int64) error {
```
Comment (wy65701436) [inline src/pkg/member/manager.go:106]: UT?

File: src/pkg/member/dao/dao.go [hunk 2]
```diff
@@ -261,3 +263,15 @@ func (d *dao) ListRoles(ctx context.Context, user *models.User, projectID int64)
 	}
 	return roles, nil
 }
+
+func (d *dao) DeleteProjectMemberByProjectID(ctx context.Context, projectID int64) error {
```
Comment (wy65701436) [inline src/pkg/member/dao/dao.go:267]: UT?

File: src/controller/event/handler/internal/project.go [hunk 3]
```diff
@@ -39,16 +40,15 @@ func (a *ProjectEventHandler) IsStateful() bool {
 
 func (a *ProjectEventHandler) onProjectDelete(ctx context.Context, event *event.DeleteProjectEvent) error {
 	log.Infof("delete project id: %d", event.ProjectID)
-	// delete tag immutable
-	err := immutable.Ctr.DeleteImmutableRuleByProject(ctx, event.ProjectID)
-	if err != nil {
+	if err := immutable.Ctr.DeleteImmutableRuleByProject(ctx, event.ProjectID); err != nil {
```
Comment (zyyw) [inline src/controller/event/handler/internal/project.go:43]: keep the comment `// delete tag immutable`, `// delete tag retention` ?

Comment (stonezdj) [reply src/controller/event/handler/internal/project.go:43]: The meaning already exists in the method name.

**Review by zyyw — APPROVED** [review]

**Pull request merged**